Make the board ID available in board and boot_out

Neradoc opened the pull request:
This implements #5105 but I am not sure of how much space it takes (for small boards), am I doing strings right ?

Additional line in boot_out.txt (there happens to be a boot.py)
```
Adafruit CircuitPython 7.0.0-beta.0-33-g802ca5425 on 2021-08-25; Raspberry Pi Pico with rp2040
Board ID:raspberry_pi_pico
boot.py output:
```
`board.ID`
```py
Adafruit CircuitPython 7.0.0-beta.0-33-g802ca5425 on 2021-08-25; Raspberry Pi Pico with rp2040
>>> import board
>>> board.ID
'raspberry_pi_pico'
```

Used the following to add to all pins.c
```py
import re
import glob

liste = glob.glob("ports/*/boards/*/pins.c")
THELINE = r"(STATIC const mp_rom_map_elem_t board_[a-z_]+\[\] = \{)"
NEWLINE = "    { MP_ROM_QSTR(MP_QSTR_ID), MP_ROM_PTR(&board_module_id_obj) },\n"

for mkfile in liste:
    board_id = mkfile.split("/")[3]
    with open(mkfile,"r") as fp:
        data = fp.read()
    if NEWLINE in data:
        continue
    if not THELINE.find(data):
        print("OH NO", board_id)
        print("    ", mkfile)
    else:
        output = re.sub(THELINE, "\\1\n" + NEWLINE, data)
        with open(mkfile,"w") as fp:
            fp.write(output)
```

microDev1 reviewed (commented):
Thanks for taking this on @Neradoc. I have one comment regarding doing this as `board.ID`.
In my opinion `board` isn't a perfect place for this as it is meant for pins and there can also be a conflict with a pin named `ID`.

In #5105, @dhalbert suggested doing this in `os.uname` and that would be my preference as well.
```python
# as a part machine 
>>> os.uname()
(sysname='rp2040', nodename='rp2040', release='7.0.0', version='7.0.0-alpha.6 2021-08-12', machine='Adafruit QT Py RP2040 (adafruit_qtpy_rp2040) with rp2040')
# or as a separate property
>>> os.uname()
(sysname='rp2040', nodename='rp2040', release='7.0.0', version='7.0.0-alpha.6 2021-08-12', machine='Adafruit QT Py RP2040 with rp2040', machine_id='adafruit_qtpy_rp2040')
```

Neradoc commented:
I chose not to add it to uname.machine because it feels wrong to have to parse it to get at the id. In this case, (no regex, promise) `machine.split("(")[1].split(")")[0]` ? I'd prefer the actual board ID be a clean separate field.

I don't know if `machine_id` is an option, it's not a C-python thing. Can we extend os.uname like that ?

An alternative mentioned in the original discussion (on circup) would be to replace uname.nodename. On Unix it's supposed to be the "network name" (though it is implementation-dependent). This would however probably not apply to Circuitpython: it would have to be set dynamically, and the network libraries are a better place for that. So maybe we can do whatever we want with it ?
C-Python doesn't really care about it either:
> Some systems truncate nodename to 8 characters or to the leading component; a better way to get the hostname is socket.gethostname() or even socket.gethostbyaddr(socket.gethostname()).

dhalbert commented:
`os.uname` is not even present on non-Unix systems (e.g., Windows). The [`platform`](https://docs.python.org/3/library/platform.html) module seems to be the preferred place for an OS-independent source of information about the machine you're running on. However, there is no field there for describing the model name of the hardware, which is really what we're looking for, e.g. "Dell Optiplex 5060", or "Circuit Playground Express".

This info is extracted from the BIOS or equivalent on PC's, via the DMI (Desktop Management Interface). But on Macs, it's completely different, of course, and more complicated. Blinka has to use a variety of ad hoc techniques to find out which board it's running on. There is really no good straightforward way. If it were ever added to CPython, it would probably be in `platform`, but I did a bunch of searching and didn't find any real discussion on this.

`os.uname().nodename` is possible, but that is also still the network name, which would be expected to diffe between two otherwise identical machines, so even though the API is compliant with CPython, the semantics is not.

Because of all this, I am inclined to say that `board` is the best place after all. I would propose that we use a lowercase `board.id` or `board.name`, or if a function call is needed, `board.id()` or `board.name()`. I would expect `board.id` to be the ID (`feather_mo_express`), and `board.name` to be the human-friendly name (`Adafruit Feather M0 Express`), which is `MICROPY_HW_BOARD_NAME`.

microDev1 commented:
> it feels wrong to have to parse it to get at the id. In this case, (no regex, promise)

I guess we can do something like `Adafruit QT Py RP2040 (id: adafruit_qtpy_rp2040) with rp2040` to make it more clear.

> I don't know if `machine_id` is an option, it's not a C-python thing. Can we extend os.uname like that ?

Good point, I totally forgot the CPython compatibility aspect there.

> An alternative mentioned in the original discussion (on circup) would be to replace uname.nodename.

`nodename` doesn't strike as something that would be containing the `id`.
